Errors for Graph Type (#341)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors/22NC4.adoc

= 22NC4

== Status description

> The label(s) `{ <<labelList>> }` are defined as both identifying and implied.

Reviewer: Does the thumb up mean you added tests on the non-docs pr for a scenario where this would be more than one label?

> The label(s) `{ <<labelList>> }` are defined as both identifying and implied.

Author: Yes

= 22NC4

== Status description
error: data exception - a label cannot be both identifying and implied. The label(s) `{ <<labelList>> }` are defined as both identifying and implied.

== Explanation
When defining a graph type, labels and relationships must be assigned clear roles.
A label cannot be both "identifying" (to uniquely identify nodes) and "implied" (automatically assigned based on other schema rules) at the same time.
Attempting to define a label as both will result in this error, as it creates ambiguity in the schema definition and node identification process.

== Example scenario
For example, try to set a graph type which defines the node element types `Person` and `Student` with the label `Person` also declared as an implied label on the `Student` node element type as follows:

[source,cypher]
----
ALTER CURRENT GRAPH TYPE SET {
    (p:Person => {name :: STRING}),
    (s:Student => :Person )
}
----

The query returns an error with GQLSTATUS 22NC4 and the status description:

[source]
----
error: data exception - a label cannot be both identifying and implied. The label(s) `Person` are defined as both identifying and implied.
----

> One way to fix this is to add another label to both nodes and capture the property type on `name` there:

Reviewer 2: Maybe we can remove one way, as it implies that we are going to give another way as well.
```suggestion
To fix this, add another label to both nodes and capture the property type on `name` there:
```

== Explanation
When defining a graph type, labels and relationships must be assigned clear roles.

> A label cannot be both "identifying" (to uniquely identify nodes) and "implied" (automatically assigned based on other schema rules) at the same time.

Reviewer: ```suggestion
A label cannot be both "identifying" (uniquely identify nodes) and "implied" (automatically assigned based on other schema rules) at the same time.
```
the `to` sounds odd to me when removing the `used`

Reviewer 2: ```suggestion
A label cannot be both "identifying" (uniquely identifies nodes) and "implied" (automatically assigned based on other schema rules) at the same time.
```
